patroni/ha.py

```diff
@@ -277,7 +277,10 @@ def update_lock(self, write_leader_optime: bool = False) -> bool:
         if write_leader_optime:
             try:
                 last_lsn = self.state_handler.last_operation()
-                slots = self.state_handler.slots()
+                slots = self.cluster.filter_permanent_slots(
+                    {**self.state_handler.slots(), slot_name_from_member_name(self.state_handler.name): last_lsn},
+                    self.is_standby_cluster(),
+                    self.state_handler.major_version)
```

Comment:
Maybe we could add a docstring to this method, taking advantage that we are "modifying" it? :)